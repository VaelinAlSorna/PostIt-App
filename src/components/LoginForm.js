import React from 'react';

class LoginForm extends React.Component {
    state = {
        name: '',
        password: ''
    };
    onNameChange = (e) => {
        const name = e.target.value;
        this.setState(() =>({
            name
        }));
    };
    onPasswordChange = (e) => {
        const password = e.target.value;
        this.setState(() =>({
            password
        }));
    };
    onSubmit = (e) => {
        e.preventDefault();
        this.props.onSubmit({
            name: this.state.name,
            password: this.state.password
        });
    }
    render () {
        return (
            <div>
                <form onSubmit={this.onSubmit}>
                    <input onChange={this.onNameChange} placeholder="Your name" type="text" value={this.state.name} required/>
                    <input onChange={this.onPasswordChange} placeholder="Password" type="password" value={this.state.password} required/>
                    <input type="submit" value="Login" />
                    {/*This checks if the error with authentication exists and if it occured on this form*/}
                    { (this.props.authFailed && this.props.authFailed.login )&& <p>{this.props.authFailed.error}</p> }
                </form>
                <p>OR</p>
                <button onClick={this.props.changeForm}>Sign Up!</button>
            </div>
        );
    };
};

export default LoginForm;