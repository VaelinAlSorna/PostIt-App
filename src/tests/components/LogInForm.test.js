import React from 'react';
import { shallow } from 'enzyme';
import LoginForm from '../../components/LogInForm';

let changeForm, onSubmit, wrapper;

beforeEach(() => {
    changeForm = jest.fn();
    onSubmit = jest.fn();
    wrapper = shallow(<LoginForm 
        authFailed={null} 
        changeForm={changeForm} 
        onSubmit={onSubmit}
    />)
})

test('should render LogInForm correctly', () => {
    expect(wrapper).toMatchSnapshot();
});

test('should call this.props.onSubmit when form is submitted', () => {
    const name = 'Mark';
    const password = 'heslo1';
    wrapper.setState({ name, password});
    wrapper.find('form').simulate('submit' ,{
        preventDefault: () => {}
    });
    expect(onSubmit).toHaveBeenLastCalledWith({name, password});
});

test('should change name on input change', () => {
    const value = 'Mark';
    wrapper.find('input').at(0).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('name')).toBe(value);
});

test('should change password on input change', () => {
    const value = 'heslo1';
    wrapper.find('input').at(1).simulate('change', {
        target: { value }
    });
    expect(wrapper.state('password')).toBe(value);
});

test('should call changeForm when clicked on OR SignUp button', () => {
    wrapper.find('button').simulate('click');
    expect(changeForm).toHaveBeenCalled();
});

test('should render error when authFailed is passed down with value', () => {
    wrapper = shallow(<LoginForm 
        authFailed={{ error: 'Wrong name or password.', login: true}} 
        changeForm={changeForm} 
        onSubmit={onSubmit}
    />)
    expect(wrapper).toMatchSnapshot();
});
