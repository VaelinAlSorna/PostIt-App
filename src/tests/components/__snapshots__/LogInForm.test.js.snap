// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render LogInForm correctly 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Your name"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <input
      type="submit"
      value="Login"
    />
  </form>
  <p>
    OR
  </p>
  <button
    onClick={[Function]}
  >
    Sign Up!
  </button>
</div>
`;

exports[`should render error when authFailed is passed down with value 1`] = `
<div>
  <form
    onSubmit={[Function]}
  >
    <input
      onChange={[Function]}
      placeholder="Your name"
      required={true}
      type="text"
      value=""
    />
    <input
      onChange={[Function]}
      placeholder="Password"
      required={true}
      type="password"
      value=""
    />
    <input
      type="submit"
      value="Login"
    />
    <p>
      Wrong name or password.
    </p>
  </form>
  <p>
    OR
  </p>
  <button
    onClick={[Function]}
  >
    Sign Up!
  </button>
</div>
`;
